### DISCUSSION TITLE
Combos not rendering?

### DISCUSSION THREAD

**Discussion by pm0u** [caksoylar/keymap-drawer#140] [Q&A] [answered]
I have a keymap that is using some helper macros from https://github.com/urob/zmk-helpers

Examples here: https://github.com/pm0u/zmk-lily58/blob/main/config/combos.dtsi

These don't seem to get rendered when using the keymap drawer: https://keymap-drawer.streamlit.app/?zmk_url=https%3A%2F%2Fgithub.com%2Fpm0u%2Fzmk-lily58%2Fblob%2Fmain%2Fconfig%2Flily58.keymap

I understand that the ones near the bottom might not be able to be resolved completely due to the weird behavior/tap dance hack - but I would have expected the first couple of fairly standard combos to work out properly (they did when defined directly in the keymap file).

This is not my usual environment so its entirely possible that there's some syntax error that is meaning this doesn't work 😅

Thanks for the great tool!

Edit
-----
Found this issue here: https://github.com/caksoylar/keymap-drawer/issues/101

I tried adding the recommended config, but still am having no luck

**Comment by pm0u** [comment, marked as answer]
Got it, i guess. It now works in the GH action... my  best understanding of the issue here is that this needs more of an actual build environment than my local provides to resolve all of this (west/zephyr/zmk... etc)

**Reply by caksoylar** [reply]
The problem is that the web app doesn't fetch the `zmk-helpers` module, whereas the build workflow does when you [use this arg](https://github.com/pm0u/zmk-lily58/blob/main/.github/workflows/draw.yml#L17). I have been meaning to add that functionality to the web app but haven't found the time yet.

Similarly for a local run, you have the have the `zmk-helpers` checked out locally and then point to its location in https://github.com/pm0u/zmk-lily58/blob/main/keymap_drawer.config.yaml#L40.

**Reply by pm0u** [reply]
ah nice - can that point to a relative directory outside of the current one?

eg - 
```
.
└── dir/
    ├── zmk-config
    └── zmk-helpers
```

**Reply by pm0u** [reply]
it seems so! excellent.

I guess I'll just have to maintain separate configs...

**Reply by caksoylar** [reply]
Sure, you can use .. or an absolute path. Note that unfortunately the relative paths are interpreted relative to your current working directory (where you issue the command), not the config file (another thing to fix at some point).

**Reply by pm0u** [reply]
didn't love the idea of maintaining 2x configs so I put a symlink in the ~`config`~ root dir and gitignored it. seems to work - now I can just leave that include line as is and it will work in CI and local. Thanks for all the help and for the great tool!

```sh
# root dir of the repository -> root dir of the helper repository
ln -s ../urob-zmk-helpers zmk-helpers
echo 'zmk-helpers' >> .gitignore
```
Edit: Had to have it all "line up" with the directory that the config points to - to keep the include line as `- "zmk-helpers/include"` this meant that the helpers symlink had to be sibling to the keymap drawer config file.
